src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -83,6 +82,18 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
 
                 IApplicationIdProvider appIdProvider = provider.GetService<IApplicationIdProvider>();
 
+                // Because of https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/943
+                // we have to touch (and create) Active configuration before initializing telemetry modules
+                // Active configuration is used to report AppInsights heartbeats
+                // role environment telemetry initializer is needed to correlate heartbeats to particular host
+                if (!string.IsNullOrEmpty(options.InstrumentationKey) &&
+                    string.IsNullOrEmpty(TelemetryConfiguration.Active.InstrumentationKey))
+                {
+                    TelemetryConfiguration.Active.InstrumentationKey = options.InstrumentationKey;
+                    TelemetryConfiguration.Active.TelemetryInitializers.Add(
+                        new WebJobsRoleEnvironmentTelemetryInitializer());
```

Comment:
We add this initializer to our TelemetryConfiguration as well, right? Do we still need to do that?

Reply:
yes, because heartbeats are reported through TelemetryConfiguration.Active: https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/943 👎

Reply:
Oh! Now I get it. I thought `WebJobsRoleEnvironmentTelemetryInitializer` was the thing that did the hearbeat. Then I realized that's ours, duh. :-)